export_as_locals.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terramate
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/madlambda/spells/errutil"
+	"github.com/mineiros-io/terramate/config"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/hcl/eval"
+	"github.com/zclconf/go-cty/cty"
+)
+
+const ErrExportAsLocalsRedefined errutil.Error = "export_as_locals attribute redefined"
+
+// ExportAsLocals represents information obtained by parsing and evaluating
+// export_as_locals blocks.
+type ExportAsLocals struct {
+	attributes map[string]cty.Value
+}
+
+// LoadStackExportAsLocals loads from the file system all export_as_locals for
+// a given stack. It will navigate the file system from the stack dir until
+// it reaches rootdir, loading export_as_locals and merging them appropriately.
+//
+// More specific definitions (closer or at the stack) have precedence over
+// less specific ones (closer or at the root dir).
+//
+// Metadata and globals for the stack are used on the evaluation of the
+// export_as_locals blocks.
+//
+// The rootdir MUST be an absolute path.
+func LoadStackExportAsLocals(rootdir string, sm StackMetadata, g *Globals) (ExportAsLocals, error) {
+	if !filepath.IsAbs(rootdir) {
+		return ExportAsLocals{}, fmt.Errorf("%q must be an absolute path", rootdir)
+	}
+
+	unEvalExport, err := loadStackExportAsLocals(rootdir, sm.Path)
```

Comment:
`unEvalExport` is funny :smile:

First, you have symbols or variables and after evaluation you get values. So maybe:
```
localVars, err := loadStackLocals(rootdir, sm.Path)
localValues, err := localVars.eval(sm, g)
```
as globals and metadata are kinds of variables.

Reply:
man I hated this uneval naming thing..but at the time I was completely stuck on not being able to think on better names, thanks :100: